numpy-stubs/__init__.pyi

```diff
@@ -326,6 +335,16 @@ class ndarray(_ArrayOrScalarCommon, Iterable, Sized, Container):
     ) -> None: ...
     def dump(self, file: str) -> None: ...
     def dumps(self) -> bytes: ...
+    @overload
+    def astype(
```

Comment:
Kind of the common theme here is adding overloads that distinguish a known dtype from a dtype-like, because in the former case we can make much stronger assertions about the output types.

Reply:
Eventually we could handle things like `dtype='float64'` too by adding an overload for `Literal['float64']`.